### PR TITLE
Add timeout in account reports

### PR DESCRIPTION
**Pull request by raimonesteve**
#158319

### CODE REVIEW

File: message.xml [hunk 1]
```diff
@@ -9,5 +9,8 @@ this repository contains the full copyright notices and license terms. -->
       <record model="ir.message" id="fiscalyear_end_date">
           <field name="text">The final posting date does not match the fiscal year "%(fiscalyear)s".</field>
       </record>
+      <record model="ir.message" id="timeout_exception">
+          <field name="text">Report computation timed out. You may consider increasing the timeout of the account reports.</field>
```
Comment (albertca) [inline message.xml]: We should suggest using Excel output, not increasing the timeout.

File: common.py [hunk 2]
```diff
@@ -2,15 +2,39 @@
...
 from trytond.tools import reduce_ids
 from trytond.transaction import Transaction
 
-__all__ = ['FiscalYear', 'Account', 'Party']
+
+class TimeoutException(Exception):
+    pass
+
+
+class TimeoutChecker:
```
Comment (albertca) [inline common.py:20]: I don't see TimeoutChecker being used anywhere.

File: common.py [hunk 3]
```diff
@@ -2,15 +2,39 @@
...
 from sql.operators import In
 
+from trytond.model import fields
 from trytond.pool import Pool, PoolMeta
 from trytond.tools import reduce_ids
 from trytond.transaction import Transaction
 
-__all__ = ['FiscalYear', 'Account', 'Party']
+
+class TimeoutException(Exception):
```
Comment (albertca) [inline common.py:16]: I don't see TimeoutException being used anywhere.

File: message.xml (hunk 1 above)
Comment (raimonesteve) [reply message.xml]: IMHO, both options...

4544fafc1bb94850810bd892b0d447bbe411b8fd

File: common.py (hunk 2 above)
Comment (raimonesteve) [reply common.py:20]: Check at:

general_ledger/general_ledger.py:15:from trytond.modules.account_reports.common import TimeoutException, TimeoutChecker
general_ledger/general_ledger.py:577:        checker = TimeoutChecker(timeout, cls.timeout_exception)

File: common.py (hunk 3 above)
Comment (raimonesteve) [reply common.py:16]: check at:

general_ledger/general_ledger.py:15:from trytond.modules.account_reports.common import TimeoutException, TimeoutChecker

```
@classmethod
def timeout_exception(cls):
    raise TimeoutException
```

File: general_ledger/general_ledger.py [hunk 4]
```diff
@@ -374,169 +387,209 @@ def _get_key_id(currentKey):
...
+            except TimeoutException:
+                raise UserError(gettext('account_reports.timeout_exception'))
+        end_prepare = datetime.now()
 
         context = Transaction().context.copy()
         context['report_lang'] = Transaction().language
         context['report_translations'] = os.path.join(
                 os.path.dirname(__file__), 'translations')
+        if timeout:
+            context['timeout'] = timeout - int((end_prepare - start_prepare).total_seconds())
```
Comment (albertca) [inline general_ledger/general_ledger.py]: timeout -> report_timeout

**Review by albertca — APPROVED** [review]

**Pull request merged**